pkg/auth/context/context.go

```diff
@@ -0,0 +1,21 @@
+package auth
+
+import "context"
+
+// SAAuthenticatedContextKey is the context key for the SAAuthenticated flag.
+var SAAuthenticatedContextKey = struct{}{}
```

Comment:
Using primitive type for context key can cause collisions (https://pkg.go.dev/context#WithValue).

I.e. defining another `var` with struct, or if someone is going to set another key in the context with the same `struct{}{}` key will override the value: https://play.golang.com/p/h-AYNP7H3xo

Another way to do this is to use typed const, or a custom type:

https://play.golang.com/p/F_C-ITYHiGv

```go
type contextKey int

const (
	SAAuthenticatedContextKey contextKey = iota
	AnotherContextKey
)
```

https://play.golang.com/p/w23oOFq601X

```go
type contextKey string

var (
	SAAuthenticatedContextKey = contextKey("SAAuthenticatedContextKey")
	AnotherContextKey         = contextKey("AnotherContextKey")
)
```

---

```suggestion
type contextKey int

const SAAuthenticatedContextKey contextKey = iota
```